Library/Homebrew/software_spec.rb

```diff
@@ -53,7 +53,7 @@ def owner=(owner)
     @resource.owner = self
     resources.each_value do |r|
       r.owner     = self
-      r.version ||= version
+      r.version ||= version.dup
```

Comment:
Also FYI, the reason that we need discard commit is otherwise we will have to redownload sources every time we install formula. (Resource version will interfere the cache file name)